docs/user-guide/lazy-loading.md

```diff
@@ -0,0 +1,94 @@
+# Lazy Loading
+
+> [!WARNING]
+> This is an experimental option and may not work as expected with all plugins.
+> The API may change without notice. > Please report any issues you encounter.
+
+> [!NOTE]
+> When the lazy loading functionality is enabled, the plugin's configuration is
+> routed to the enabled lazy provider. Options mentioned will be in relation to
+> the `${namespace}.${name}.lazyLoad` context.
+
+## Supported Lazy Loading Providers
+
+It is recommended to become familiar with lazy loading strategies supported by
+your provider in their upstream documentation and the corresponding plugin's
+support.
+
+Currently we only support **one** lazy loader:
+
+-lz.n: [`plugins.lz-n`](/plugins/lz-n/index.md) -
+[upstream docs](https://github.com/nvim-neorocks/lz.n?tab=readme-ov-file#plugin-spec)
+
+## Enabling Lazy Loading Per Plugin
+
+There are a couple different ways to enable lazy loading for a particular
+plugin.
+
+- `lazyLoad.enable` set to `true`.
+- `lazyLoad.settings` containing some configuration.
+
+If you are just wanting to store potential configuration without enabling it,
+you can explicitly disable it setting `lazyLoad.enable = false`.
```

Comment:
Continuing from https://github.com/nix-community/nixvim/pull/2630#discussion_r1884715931

I wonder if this would be clearer without mention that `enable` can be used to enable... Seeing is you should never really need to enable without also configuring `settings`, right?

Therefore I'd suggest we remove the list and replace the first para with an explanation about what `settings` is and the fact that configuring it will auto-enable.

The final para already covers use cases for `enable = false`, so I think that is ok as-is?

Reply:
That's fair, I was about to add a mention about requiring the settings and then realized it felt weird to even mentioning enabling, by itself.